### PR TITLE
Lock nokogiri to 1.7.x for Ruby 2.1

### PR DESCRIPTION
**Pull request by parkr**
Fails our AppVeyor tests: https://ci.appveyor.com/project/jekyll/jekyll/build/1722/job/3vlp2o2i69uw6n2p

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -22,7 +22,8 @@ group :test do
   gem "cucumber", "~> 2.1"
   gem "jekyll_test_plugin"
   gem "jekyll_test_plugin_malicious"
-  gem "nokogiri"
+  # nokogiri v1.8 does not work with ruby 2.1 and below
+  gem "nokogiri", RUBY_VERSION > "2.1" ? "~> 1.7" : "~> 1.7.0"
```
Comment (DirtyF) [inline Gemfile]: Shouldn't it be something like:
`gem "nokogiri",  RUBY_VERSION < "2.1" ? "1.6" : "1.7"`

Comment (ashmaroli) [inline Gemfile]: no @DirtyF , it should instead be:
```ruby
gem "nokogiri", RUBY_VERSION >= "2.2" ? "~> 1.7" : "~> 1.7.0"
```
`> 2.1` will include `2.1.1`, `2.1.2` etc

Comment (pathawks) [inline Gemfile]: `> "2.1"` is equivalent to `>= "2.2"`

?

Comment (ashmaroli) [inline Gemfile]: no it isn't

Comment (ashmaroli) [inline Gemfile]: `> 2.1` will include `2.1.1`, `2.1.2`... `2.2.x`, `2.3.x`, `2.4.x`
`>= 2.2` will include only `2.2.0`, `2.2.1`,..., `2.4.x`

Comment (pathawks) [inline Gemfile]: Is it different practically or only semantically? 😕

Comment (ashmaroli) [inline Gemfile]: both.. Appveyor still fails with the last commit because, its still installing `1.8.0` instead..

Comment (pathawks) [inline Gemfile]: TIL :beers:

**Comment by parkr** [comment]
💚 

@jekyllbot: merge +dev

**Pull request merged**